File: backend/src/main/java/com/example/warsaw_salons/domain/salon/Salon.java
package com.example.warsaw_salons.domain.salon;

import jakarta.persistence.*;
import lombok.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Table(name = "salons")
public class Salon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;
    private String district;

    private String phoneNumber;
    private String website;
    private String servicesOffered;
    private String priceRange;
    private Double rating;
    private Integer numberOfReviews;
}

File: frontend/src/types/salon.ts
export interface Salon {
    id: number;
    name: string;
    district: string;
    priceRange: string;
    rating: number;
    address?: string;
    phoneNumber?: string;
    website?: string;
    servicesOffered?: string;
    numberOfReviews?: number;
}

export interface SalonMutateRequest {
    name: string;
    address: string;
    district: string;
    phoneNumber: string;
    website: string;
    servicesOffered: string;
    priceRange: string;
    rating: number;
    numberOfReviews: number;
}

export interface PageResponse<T> {
    content: T[];
    totalPages: number;
    totalElements: number;
    size: number;
    number: number;
}